packages/libs/error-stack/src/report.rs

```diff
@@ -205,13 +234,94 @@ impl<C> Report<C> {
         #[cfg(all(nightly, any(feature = "std", feature = "spantrace")))]
         drop(provider);
 
-        Self::from_frame(
-            Frame::from_context(context, Location::caller(), None),
-            #[cfg(all(nightly, feature = "std"))]
-            backtrace,
-            #[cfg(feature = "spantrace")]
-            span_trace,
-        )
+        // We allow `unused_mut` due to the fact that certain feature configurations will require
+        // this to be mutable (backtrace and spantrace overwrite the source)
+        #[allow(unused_mut)]
+        let mut source: Box<[Frame]> = Box::new([]);
+
+        #[cfg(all(nightly, feature = "std"))]
+        if let Some(backtrace) = backtrace {
+            let frame = Frame::from_attachment(backtrace, Location::caller(), source);
```

Comment:
We better use the provided method `.attach(backtrace)`. This ensures, that the root frame is *always* a `Context`.